hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -146,7 +146,7 @@ protected ClientDelegatingFuture getAsyncInternal(Data dataKey, ExpiryPolicy exp
         try {
             long reservationId = nearCache.tryReserveForUpdate(dataKey);
             ClientDelegatingFuture future = super.getAsyncInternal(dataKey, expiryPolicy, callback);
```

Comment:
Shouldn't it be something like this? So we only have 1 execution callback instead of 2.
```
GetAsyncCallback getAsyncCallback = new GetAsyncCallback(dataKey, reservationId, callback);
ClientDelegatingFuture future = super.getAsyncInternal(dataKey, expiryPolicy, getAsyncCallback);
```